src/components/Drawer/components/Drawer.scss

```diff
@@ -0,0 +1,199 @@
+@use '../../variables';
+
+$block: '.#{variables.$ns}drawer';
+
+#{$block} {
+    pointer-events: none;
+    outline: none;
+
+    --_--item-shadow-default: 0 1px 5px 0 var(--g-color-sfx-shadow);
+    --_--item-position: fixed;
+
+    --_--resizer-width: 8px;
+    --_--resizer-color: var(--g-color-base-generic);
+    --_--resizer-handle-color: var(--g-color-line-generic);
+    --_--resizer-handle-color-hover: var(--g-color-line-generic-hover);
+    --_--resizer-z-index: 100;
+
+    --_--veil-background-color: var(--g-color-sfx-veil);
+
+    &_hide-veil {
+        --_--item-shadow: var(--g-drawer-item-shadow, var(--_--item-shadow-default));
+    }
+
+    &__item {
+        position: var(--g-drawer-item-position, var(--_--item-position));
+        z-index: var(--g-drawer-item-z-index, 110);
+        inset-inline-start: var(--g-drawer-item-left-offset, 0);
+        inset-block: var(--g-drawer-item-top-offset, 0) 0;
+
+        height: 100%;
+
+        background-color: var(--g-color-base-background);
+
+        pointer-events: initial;
+        box-shadow: var(--_--item-shadow, none);
+        outline: none;
+        will-change: transform;
+
+        &_direction_right {
+            inset-inline: auto 0;
+        }
+
+        &_direction_top {
+            inset-block-end: auto;
+            inset-inline-end: 0;
+        }
+
+        &_direction_bottom {
+            inset-block-start: auto;
+            inset-inline-end: 0;
+        }
+
+        &_disable-portal {
+            --_--item-position: static;
```

Comment:
`disablePortal` should behave like in `Modal` component. It shouldn't render in place